database.c

```diff
@@ -307,17 +308,25 @@ static const char *current_(const char *profile, const char *key)
   return res;
 }
 
-/* ========================================================================= *
- * FIXME: horrible $HOME/.profiled access hacks
- * ========================================================================= */
-
 static const char *datadir(void)
+{
+  const char *confdir = (char*)g_get_user_config_dir();
```

Comment:
Existing thing, but since playing with glib, could consider fully doing gchar and e.g. g_snprintf(), though that might leak into other parts of the code here.